core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -187,47 +200,74 @@ private[history] class FsHistoryProvider(conf: SparkConf) extends ApplicationHis
         }
         .flatMap { entry =>
           try {
-            Some(replay(entry, new ReplayListenerBus()))
+            Some((getModificationTime(entry).get, entry))
           } catch {
             case e: Exception =>
               logError(s"Failed to load application log data from $entry.", e)
               None
           }
         }
-        .sortWith(compareAppInfo)
+        .sortWith(_._1 >= _._1)
+        .map { case (_, file) => file }
```

Comment:
I think you can replace these three cases (the flatMap, sortWith, and map) with a single one.
```
val logInfos: Seq[FileStatus] = statusList
  .filter { ... }
  .sortWith { case (entry1, entry2) =>
    val mod1 = getModificationTime(entry1).getOrElse(-1)
    val mod2 = getModificationTime(entry1).getOrElse(-1)
    mod1 >= mod2
  }
```